### ISSUE TITLE
Wrong input and models for time series forecasting

### ISSUE THREAD

**Issue by nick-janssen** [marthawhite/Histogram_loss#25]
Hello,

while using your repository for time series prediction on the ETTm1 dataset, I noticed two aspects that I don’t fully understand:

- train_step function in HistModel / HLGaussian:
I saw that a train_step function is implemented in the HistModel, but it doesn’t seem to be called during training. Could you clarify how the cross-entropy loss is applied, given that the training loop only appears to use the MSE loss?

- Dataset creation and prediction length:
When building the dataset for ETTm1, it looks like the prediction_length parameter isn’t used. As a result, only a single future time step per training sample is generated, rather than the 96 steps described in your paper:
`ys = keras.utils.timeseries_dataset_from_array(df_targets[:,-1], None, 1, batch_size=None)`
Is this intentional, or should the dataset construction include multiple future steps?

I’m not very experienced with Keras/TensorFlow, so I’d really appreciate any clarification on these points.

Thanks a lot for your work on this project!

Best,
Nick

**Comment by marthawhite** [comment]
Hi Nick,

Thanks for raising this issue. I know Esraa (cc'ed) is looking into it, and
she can respond here.

-Martha

**Comment by esraaelelimy** [comment]
Hi Nick, sorry for the late response. Thanks for pointing out this issue, it was indeed a bug in those experiments. We have updated the experiments and added new ones, please check this [branch](https://github.com/marthawhite/Histogram_loss/tree/updated_timeseries) for the updated experiments. A new version of the paper will also be out soon with the updated experiments.
